src/main/java/com/google/api/codegen/util/ProtoParser.java

```diff
@@ -86,6 +89,51 @@ public ProtoParser(boolean enableProtoAnnotations) {
     }
   }
 
+  /* Return the name of the field representing the header parameter. */
+  public ImmutableSet<String> getHeaderParams(Method method) {
```

Comment:
Minor: would just specifying method signature as returning `Set<String>` (when the actual returned object will still be `ImmutableSet<String>`) be enought here? The best practice is to always try to declare in the method signature the most generic ("generic" in its regular meaning, I do not mean java generics here) input/return type(s) (but the one which still "fit"). The more generic the declaration is the less commitment the implementor/maintainer of the method makes, thus it becomes easier to modify/update the method in the future (though that should not affect the usability of the method, otherwise everythign will be returning and accepting `Object` type).

Reply:
`Set<>` isn't guaranteed to any order; I want to assure the method callers that the return type has insertion-order, which is what the `ImmutableSet` type conveys.